src/pkgcheck/checks/codingstyle.py

```diff
@@ -903,3 +903,71 @@ def feed(self, pkg):
                         yield RedundantDodir(
                             cmd.group('cmd'), line=dodir.group('call'),
                             lineno=lineno - 1, pkg=pkg)
+
+
+class UnquotedVariable(results.LineResult, results.Style):
+    """Variable is unquoted, but should be quoted"""
+
+    def __init__(self, variable, **kwargs):
+        super().__init__(**kwargs)
+        self.variable = variable
+
+    @property
+    def desc(self):
+        return f"{self.lineno}: Variable {self.variable} should be quoted"
+
+
+class UnquotedVariablesCheck(Check):
+    """Scan ebuild for variables that should be quoted like D, FILESDIR, etc."""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset([UnquotedVariable])
+
+    message_commands = (
+        "die", "echo", "eerror", "einfo", "elog", "eqawarn", "ewarn"
+    )
+    var_names = (
+        "D", "DISTDIR", "FILESDIR", "S", "T", "ROOT", "BROOT", "WORKDIR",
+        # EAPI=3/Prefix vars
```

Comment:
I think this comment is kind of pointless (know it came from repoman) as _above_ it we have the new (EAPI 7+) `BROOT` anyway

Reply:
I think mgorny's covered the remaining stuff from https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-10900011.1.